torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,202 @@
+  if (inputs.size() < min_size) {
+      ss << "Failed to save extra arguments for flops compuation of op "
+         << op_name
+         << ", min size: " << min_size
+         << ", actual size: " << inputs.size();
+      TORCH_WARN(ss.str());
+      return false;
+  }
+  for (auto index : should_be_tensor) {
+    if (!inputs[index].isTensor()) {
+      ss << "Failed to save extra arguments for flops compuation of op "
+         << op_name
+         << ", input[" << index
+         << "] must be a tensor.";
+      TORCH_WARN(ss.str());
+      return false;
+    }
+  }
+  return true;
+}
+
+std::unordered_map<std::string, c10::IValue> saveExtraArgs(const at::RecordFunction& fn) {
+    // foe specific types of fn, return the saved extra args for computing flops
+    std::unordered_map<std::string, c10::IValue> map;
+    std::vector<c10::IValue> inputs = fn.inputs();
+    std::string fname(fn.name().str());
+
+    if (inputs.empty()) {
+      // Input shape is unavailable, return empty map
+      return map;
+    }
+
+    if (fname == "aten::conv2d") {
+      std::vector<int> tensors{0, 1};
+      bool check = validateInput(fname, CONV2D_GROUPS + 1, inputs, tensors);
+      if (!check) {
+        return map;
+      }
+
+      at::Tensor input = inputs[0].toTensor();
+      at::Tensor weight = inputs[1].toTensor();
+      if (weight.sizes().size() != 4) {
+        TORCH_WARN("Failed to compute flops for op aten::conv2d because it requires a 4D kernel tensor.");
+        return map;
+      }
+      map["input_size"] = at::IValue(input.sizes());
+      map["weight_size"] = at::IValue(weight.sizes());
+      map["stride"] = inputs[CONV2D_STRIDE];
+      map["padding"] = inputs[CONV2D_PADDING];
+      map["dilation"] = inputs[CONV2D_DILATION];
+      map["groups"] = inputs[CONV2D_GROUPS];
+    } else if (fname == "aten::mm") {
+      std::vector<int> tensors{0, 1};
+      bool check = validateInput(fname, 2, inputs, tensors);
+      if (!check) {
+        return map;
+      }
+
+      at::Tensor left = inputs[0].toTensor();
+      at::Tensor right = inputs[1].toTensor();
+      map["mat1_size"] = at::IValue(left.sizes());
+      map["mat2_size"] = at::IValue(right.sizes());
+    } else if (fname == "aten::mul.Tensor") {
+      std::vector<int> tensors{0};
+      bool check = validateInput(fname, 1, inputs, tensors);
+      if (!check) {
+        return map;
+      }
+
+      at::Tensor mat = inputs[0].toTensor();
+      map["mat_size"] = at::IValue(mat.sizes());
+    } else if (fname == "aten::add.Tensor") {
+      std::vector<int> tensors{0};
+      bool check = validateInput(fname, 1, inputs, tensors);
+      if (!check) {
+        return map;
+      }
+
+      at::Tensor mat = inputs[0].toTensor();
+      map["mat_size"] = at::IValue(mat.sizes());
+    }
+
+    return map;
+}
+
+uint64_t computeFlops(const std::string &op_name, const std::unordered_map<std::string, c10::IValue> &extra_args) {
+    if (op_name == "aten::conv2d") {
```

Comment:
Instead of having numerous try /catches for missing argument, wouldn't be easier to validate input args at the beginning? (Or have a global try/catch that would report that it failed to find mandatory argument for op_name?)

Reply:
I have refactored the code to validate input args at the beginning of each branch. This gives more debugging information than global try/catch block when failure happens.